samcli/lib/bootstrap/nested_stack/nested_stack_manager.py

```diff
@@ -0,0 +1,116 @@
+"""
+nested stack manager to generate nested stack information and update original template with it
+"""
+import logging
+import os
+from copy import deepcopy
+from typing import Dict, cast
+
+from samcli.commands._utils.template import move_template
+from samcli.lib.bootstrap.nested_stack.nested_stack_builder import NestedStackBuilder
+from samcli.lib.build.app_builder import ApplicationBuildResult
+from samcli.lib.providers.provider import Stack
+from samcli.lib.providers.sam_function_provider import SamFunctionProvider
+from samcli.lib.utils.packagetype import ZIP
+from samcli.lib.utils.resources import AWS_SERVERLESS_FUNCTION, AWS_LAMBDA_FUNCTION
+
+LOG = logging.getLogger(__name__)
+
+# Resource name of the CFN stack
+NESTED_STACK_NAME = "AwsSamAutoDependencyLayerNestedStack"
+
+# Resources which we support creating dependency layer
+SUPPORTED_RESOURCES = {AWS_SERVERLESS_FUNCTION, AWS_LAMBDA_FUNCTION}
+
+# Languages which we support creating dependency layer
+SUPPORTED_LANGUAGES = ("python", "nodejs", "java")
+
+
+def generate_auto_dependency_layer_stack(
```

Comment:
A bit of long function. Are we able to break it down into multiple ones?

Reply:
Agreed, let me move validations into different one(s)